server/src/main/java/io/crate/metadata/GeneratedReference.java

```diff
@@ -57,8 +57,6 @@ private GeneratedReference(Reference ref,
                                String formattedGeneratedExpression,
                                Symbol generatedExpression) {
         assert generatedExpression != null : "GeneratedExpression is required";
-        assert generatedExpression.valueType().equals(ref.valueType())
-            : "The type of the generated expression must match the valueType of the `GeneratedReference`";
```

Comment:
Shouldn't this condition still hold true?

Reply:
Yes, totally, thanks. We must add references of the object's inner type based on the generated reference type, the same we do if a column as an object type with inner types. Done by https://github.com/crate/crate/pull/17139/commits/2e393e359ae06ad9f419e9fd31639670ae493e73.